grub-core/lib/gpt.c

```diff
@@ -481,9 +518,8 @@ grub_gpt_read_entries (grub_disk_t disk, grub_gpt_t gpt,
       goto fail;
     }
 
-  grub_free (gpt->entries);
```

Comment:
I don't know what the underlying entries object is, but it seems like you're missing a free by calling `grub_gpt_read_entries(..., gpt, ... &gpt->entries ...)` - why doesn't this leak?